src/docs/product/relay/best-practices.mdx

```diff
@@ -0,0 +1,65 @@
+---
+title: Self-hosting - Best Practices
+sidebar_order: 5
+---
+
+This page contains some advices and best practices when self-hosting external Relays, i.e. Relays that run on your hardware and forward events to `sentry.io`.
+
+## General Considerations
+
+* While Relay can be built from source, the recommended way to run Relay is via the officially provided Docker image (`getsentry/relay`), which can be found on [DockerHub]. The images are tagged with their Git revision identifier.
+
+* We recommend to run at least two Relay instances (containers) with a reverse proxy (such as [HAProxy] or [Nginx]) in front of them for improved availability and to simplify Relay updates.
+
+* To monitor your Relay setup, you can configure [Logging] and [Metrics].
+
+## System Requirements and Recommendations
+
+- CPU
+    - x86-64 (amd64) CPU architecture
+    - Relay is a multi-threaded application that tries to leverage all available CPU cores, thus it is highly recommended to run Relay on multi-core CPUs. If your setup is expected to handle more than 100 requests per second, it is recommended to run Relay on at least 4 CPU cores.
+        - By default, every Relay instance will use the total number of available cores to adjust the sizes of its thread pools. `limits.max_thread_count` configuration option can be set to adjust this behavior.
+- Memory
+    - At least 2GB of RAM per Relay container is recommended.
+- Storage
+    - Relay does not need any disk storage at the moment.
+
+## Example Configuration
+
+This example configuration contains a few important values that can be tweaked to change Relay's concurrency settings and buffering behavior.
+
+```yaml
+---
+relay:
+  upstream: https://o<YOUR_ORG_ID>.ingest.sentry.io
+  host: 0.0.0.0
+  port: 3000
+logging:
+  level: info
+  format: json
+metrics:
+  statsd: 127.0.0.1:8126
+  prefix: relay
+limits:
+  # Base size of various internal thread pools. Defaults to the number of logical CPU cores
+  max_thread_count: 8
+  # Number of concurrent upstream requests for event submission
+  max_concurrent_requests: 500
+cache:
+  # How many events relay buffers before it starts dropping new events
+  event_buffer_size: 10000
+  # How long a project configuration can still be used after having expired (in seconds).
+  # This might help when the upstream is unreachable (e.g., in case of network issues).
+  project_grace_period: 3600
+  # The cache timeout for buffered events before dropping them (in seconds)
+  event_expiry: 1800
```

Comment:
_nit_: Should we maybe remove them here and ensure we have sensible defaults? While the event buffer size totally depends on load (and ideally can be inferred in future), customers' expectation is probably that grace periods and expiry just work out-of-the-box. WDYT?

Reply:
I'm +1000 for having sensible defaults that we don't have to tweak for every new event pattern.

Reply:
In that case you can remove. The current defaults are sensible for external Relays. It's only our PoPs and internal Relays that need overriding.